include/dlpack/dlpack.h

```diff
@@ -89,6 +100,15 @@ typedef enum {
   kDLWebGPU = 15,
   /*! \brief Qualcomm Hexagon DSP */
   kDLHexagon = 16,
+
+  /*!
+   * \brief The end mark of DLDevice type flag.
+   * This is a helper flag to help dependent packages to check possible extensions.
+   *
+   * The dependent package should be static_assert(kExtendedFlag > kDLDeviceTypeEnd);
+   * to ensure that version update do not override existing flags.
```

Comment:
sometimes depenendent package can try to store extended device that are not yet part of DLPack (which can be included later). So this is just providing convenience for suhc packages

Reply:
Hmmmm, but that seems tricky in a dynamically linked world?  I cannot do static-asserts otherwise and would have to trust a version check.

But in that case, getting an older version, I would have to check that the device actually makes sense in the *old* version, even though I compile against the new version!

Can we say that "extended" versions MUST NOT be shared in a dynamically linked world like Python?  Or alternatively, explicitly block off an "extended" area?

Reply:
> sometimes depenendent package can try to store extended device that are not yet part of DLPack

Would it not be better to set `kDLDeviceTypeEnd` to a really large integer to ensure that there are guaranteed to be no clashes? Like sort of reserved range for vendor-specific extensions that haven't been standardized yet.

Reply:
I will suggest sacrificing the top bit for this purpose and making that explicit.

Reply:
I realize that there are different opinions on this one and it is not necessarily closely tied with this proposal updat. As a result, I will remove it for now.